code/modules/mob/living/carbon/alien/humanoid/death.dm

```diff
@@ -19,4 +19,11 @@
 		if(istype(node)) // just in case someone would ever add a diffirent node to hivenode slot
 			node.queen_death()
 
-	return ..()
+	. = ..()
+
+	SSshuttle.clearHostileEnvironment(src)
+	if(!mind)
+		return
+	var/datum/antagonist/xeno/xeno_datum = mind.has_antag_datum(/datum/antagonist/xeno)
+	if(xeno_datum)
+		xeno_datum.xeno_team.queen_deaths++
```

Comment:
Any reason to do this after calling parent? The parent call may result in the mob being gibbed and deleted, which could cause this code not to run.